hybrid/SampleApps/NoteSync/NoteSync/SFContentSoqlSyncTarget.m

```diff
@@ -0,0 +1,332 @@
+}
+
+-(void)parserDidEndDocument:(NSXMLParser *)parser
+{
+    self.completeBlock(self.records, self.totalSize, self.queryLocator);
+}
+
+-(void)parser:(NSXMLParser *)parser parseErrorOccurred:(NSError *)parseError
+{
+    NSLog(@"%@", [parseError localizedDescription]);
+}
+
+-(void)parser:(NSXMLParser *)parser didStartElement:(NSString *)elementName namespaceURI:(NSString *)namespaceURI qualifiedName:(NSString *)qName attributes:(NSDictionary *)attributeDict
+{
+    
+    if ([elementName isEqualToString:RESULT]) {
+        self.inResult = YES;
+    }
+    else if ([elementName isEqualToString:RECORDS]) {
+        self.inRecord = YES;
+        self.record = [NSMutableDictionary new];
+    }
+    self.currentElement = elementName;
+}
+
+-(void)parser:(NSXMLParser *)parser didEndElement:(NSString *)elementName namespaceURI:(NSString *)namespaceURI qualifiedName:(NSString *)qName{
+    
+    if ([elementName isEqualToString:RESULT]) {
+        self.inResult = NO;
+    }
+    else if (self.inResult && [elementName isEqualToString:RECORDS]){
+        self.inRecord = NO;
+        [self.records addObject:self.record];
+    }
+    else if (self.inResult && [elementName isEqualToString:DONE]) {
+        self.queryDone = [self.foundValue boolValue];
+    }
+    else if (self.inResult && [elementName isEqualToString:QUERY_LOCATOR]){
+        self.queryLocator = self.queryDone ? nil : self.foundValue;
+    }
+    else if (self.inResult && [elementName isEqualToString:SIZE]) {
+        self.totalSize = [self.foundValue integerValue];
+    }
+    else if (self.inRecord && [elementName hasPrefix:SF]) {
+        NSString* attributeName = [elementName substringFromIndex:[SF length]];
+        self.record[attributeName] = self.foundValue;
+    }
+    self.foundValue = [NSMutableString new];
+}
+
+-(void)parser:(NSXMLParser *)parser foundCharacters:(NSString *)string{
+    [self.foundValue appendString:string];
+}
+
+@end
+
+@interface SFSoapSoqlRequest : SFRestRequest
+
+@property (nonatomic, strong) NSString *query;
+@property (nonatomic, strong) NSString *queryLocator;
+
+- (id)initWithQuery:(NSString*) query;
+- (id)initWithQueryLocator:(NSString*) queryLocator;
+
+@end
+
+@implementation SFSoapSoqlRequest
+
+- (id)initWithQuery:(NSString*)query
+{
+    self = [super init];
+    if (self) {
+        self.method = SFRestMethodPOST;
+        self.path = SOAP_PATH;
+        self.endpoint = SOAP_ENDPOINT;
+        self.parseResponse = NO;
+        self.query = query;
+        self.queryLocator = nil;
+        
+    }
+    return self;
+}
+
+- (id)initWithQueryLocator:(NSString*)queryLocator
+{
+    self = [super init];
+    if (self) {
+        self.method = SFRestMethodPOST;
+        self.path = @"";
+        self.endpoint = SOAP_ENDPOINT;
+        self.parseResponse = NO;
+        self.query = nil;
+        self.queryLocator = queryLocator;
+    }
+    return self;
+}
+
+
+- (SFNetworkOperation*) send:(SFNetworkEngine*) networkEngine {
```

Comment:
Maybe with network sdk we will be able to pass any http entity for the body (like on Android)?